main.cpp

```diff
@@ -51,22 +51,322 @@ If you need inspiration for what to write, take a look at previously approved st
+    return false;
+}
 /*
  UDT 2:
  */
+struct Fighter 
+{
+    Fighter();
+    ~Fighter();
 
+    string weapon;
+    char signet;
+    string armor;
+    float money;
+    struct Moves {
+        Moves();
+        ~Moves();
+        string bestMove = "swing sword";
+        string worstMove = "seduce enemy";
+        bool canBeCalm = false;
+        bool fightsDrunk = true;
+        bool smellsBad = true;
+    };
+    string fight(string weaponUsed);
+    int block(int speed);
+    int drink(int stamina);
+};
+
+Fighter::Fighter() :
+weapon("Sword"),
+signet('L'),
+armor("none"),
+money(0.0f)
+{
+    cout <<"CONSTRUCTING Fighter with" << weapon  << endl;
+}
+
+Fighter::~Fighter()
+{
+    cout << "DESTRUCTING Fighter" << endl;
+}
+
+string Fighter::fight(string weaponUsed)
+{
+    if (weaponUsed != "fists")
+    {
+        cout << "Fighter fights with " << weaponUsed << endl;
+        return "Fighter fights with " + weapon;
+    }
+    cout << "Fighter fights with fists" << endl;
+    return "Fighter fights with fists";
+}
+
+int Fighter::block(int speed)
+{
+    cout << "Fighter blocks with a speed of " << speed << endl;
+    return speed;
+}
+
+int Fighter::drink(int stamina)
+{
+    if (stamina > 12)
+    {
+        cout << "Fighter manages to drink another round!";
+        return 0;
+    }
+    cout << "Unable to hold his drink the Fighter collapses" << endl;
+    return 1;
+}
 /*
  UDT 3:
  */
+struct Preistess 
+{
+    Preistess();
+    ~Preistess();
+    string favoriteFood;
+    string leastFavoriteThing;
+    int crystals;
+    int illusions;
+    string home;
+    int castGrowSpell(int numPlants = 1);
+    string castBanish(string thingToBeBanished);
+    int castIllusions(int numIllusions = 1);
+};
+
+Preistess::Preistess() :
+favoriteFood("Salad"),
+leastFavoriteThing("Bad Smells"),
+crystals(12),
+illusions(20),
+home("Tree house")
+{
+    cout << "CONSTRUCTING Preistess with " << crystals << " ctrystals"  << endl;
+}
+
+Preistess::~Preistess()
+{
+    cout << "Deconstructing Preistess" << endl;
+}
+
+int Preistess::castGrowSpell(int numPlants)
+{
+    for (int i = 0; i < Preistess::crystals; i += 1)
+    {
+        numPlants += 1;
+    }
+    cout << "Priestess casts grow and " << numPlants << " plants grow immediately" << endl;
+    return numPlants;
+}
 
+string Preistess::castBanish(string thingToBeBanished)
+{
+    if (thingToBeBanished != Preistess::leastFavoriteThing)
+    {
+        cout << "Preistess casts banish and " << thingToBeBanished << " are banished for 2 rounds" << endl;
+        return "praretess casts banish and " + thingToBeBanished + " are banished for 2 rounds";
+    }
+    cout << "Preistess casts banish and " << Preistess::leastFavoriteThing << " are banished for 2 rounds" << endl;
+        return "preistess casts banish and " + Preistess::leastFavoriteThing + " are banished for 2 rounds";
+}
+
+int Preistess::castIllusions(int numIllusions)
+{
+    int index = Preistess::crystals;
+    while (index > 0)
```

Comment:
Do not use fully-qualified names when accessing member variables.
Only use the Fully-qualified name when accessing `static` member variables and functions
- [ ] revise